tests/components/dsmr/test_sensor.py

```diff
@@ -349,6 +350,29 @@ async def test_connection_errors_retry(hass, monkeypatch, mock_connection_factor
     assert first_fail_connection_factory.call_count >= 2, "connecting not retried"
 
 
+async def test_connection_abort_error_retry(hass, monkeypatch, mock_connection_factory):
+    """Connection should be retried on connection abort error during setup."""
+    (connection_factory, transport, protocol) = mock_connection_factory
+
+    config = {"platform": "dsmr", "reconnect_interval": 0}
+
+    # override the mock to have it fail the first time and succeed after
+    first_fail_connection_factory = tests.async_mock.AsyncMock(
+        return_value=(transport, protocol),
+        side_effect=chain([OSError(errno.ECONNABORTED)], repeat(DEFAULT)),
+    )
+
+    monkeypatch.setattr(
+        "homeassistant.components.dsmr.sensor.create_dsmr_reader",
+        first_fail_connection_factory,
+    )
+    await async_setup_component(hass, "sensor", {"sensor": config})
+
+    # wait for sleep to resolve
+    await hass.async_block_till_done()
+    assert first_fail_connection_factory.call_count >= 2, "connecting not retried"
+
+
```

Comment:
Perhaps it's better to adapt `test_reconnect` instead of making a new test? This new test makes `test_reconnect` a bit unnecessary as `TimeoutError` is a subclass of `OSError`. Both tests trigger the same logic.

Reply:
That test looks a bit different than this test.

Pytest supports parametrization if the differences can be extracted to parameters.
https://docs.pytest.org/en/stable/parametrize.html

I'm not sure if that's applicable here.

Reply:
I meant `test_connection_errors_retry`. You are right, `test_reconnect` does something different.

Reply:
The parametrization seems to hit a sweet spot. No duplicate testing code but still being able to test multiple error types. I do think it's best to test individual errors (not all of them) to make sure it keeps working correctly after refactoring